File: src/main/java/bookmytrip/Entity/Restaurant.java
package bookmytrip.Entity;

import java.util.Set;

import javax.persistence.*;
import javax.validation.constraints.*;

import lombok.*;

@Entity
@Getter @Setter
@PrimaryKeyJoinColumn(name = "restaurant_id")
public class Restaurant extends Entry {
	
	@ManyToMany
	@JoinTable(
			name = "restaurant_to_cuisine",
			joinColumns = @JoinColumn(name = "restaurant_id"),
			inverseJoinColumns = @JoinColumn(name = "cuisine_id"))
	private Set<Cuisine> cuisines;
	
	@NotNull
	@Min(1) @Max(3)
	@Column(nullable = false)
	private Integer priceLevel;	
}

File: frontend/src/app/entity/restaurant.ts
import { Cuisine } from './cuisine';
import { Contact } from './contact';
import { Review } from './review';

export interface Restaurant {
    id: number;
    name: string;
    contact: Contact;
    reviews: Review[];
    avrgRating: number;
    numOfReviews: number;

    cuisines: Cuisine[];
    priceLevel: number;
}